lib/improver/nbhood/use_nbhood.py

```diff
@@ -156,3 +159,133 @@ def process(self, cube, mask_cube):
             cube, concatenated_cube,
             exception_coordinates=exception_coordinates)
         return concatenated_cube
+
+
+class CollapseMaskedNeighbourhoodCoordinate(object):
+
+    """
+    Plugin for collapsing the coordinate the mask was applied to after
+    masked neighbourhood processing.
+
+    Takes into account the result from the neighbourhood processing to
+    adjust the weights between the bands in the coordinate for the points
+    where the were no points within a neighbourhood for and a non-zero
+    weighting.
+    """
+
+    def __init__(self, coord_masked, weights=None):
+        """
+        Initialise the class.
+
+        Args:
+            coord_masked (string):
+                String matching the name of the coordinate that has been used
+                for masking.
+
+        Keyword Args:
+            weights (cube):
+                A cube from an ancillary file containing the weights for each
+                point in the coord_masked at each grid point. Only two points
+                in coord_masked can have a non-zero weight for each grid-point,
+                i.e. we are only weighting between two adjacent bands in the
+                neighbourhood output for each gridpoint.
+                Should have the coordinates coord_masked, x and y.
+                Default is None which equal weights for each band in the mean
+                used to collapse the chosen coordinate.
+                The weights cube can be masked, and this mask will be retained,
+                and will be present in the output.
+.
+        """
+        self.coord_masked = coord_masked
+        self.weights = weights
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<ApplyNeighbourhoodProcessingWithAMask: '
+                  'coord_masked: {}, weights: {}>')
+        return result.format(self.coord_masked,
+                             self.weights)
+
+    def reweight_weights(self, nbhood_cube):
```

Comment:
How about "renormalize_weights()"?  For me that would be clearer in describing the functionality.

Reply:
Done

Reply:
Done